### PR TITLE
Update RAND_bytes_ex() and RAND_priv_bytes_ex() to accept a size_t length.

### PR DESCRIPTION
**Pull request by paulidale**
The older RAND_bytes() and RAND_priv_bytes() use an `int` for their size.  This corrects that API mistake.

Interestingly, almost all instances where an `int` was passed, range checked the `int` before passing.  The few remaining cases where this didn't happen have been corrected.

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: ssl/statem/statem_clnt.c [hunk 1]
```diff
@@ -2853,7 +2853,7 @@ static int tls_construct_cke_rsa(SSL *s, WPACKET *pkt)
     pms[0] = s->client_version >> 8;
     pms[1] = s->client_version & 0xff;
     /* TODO(size_t): Convert this function */
```
Comment (FdaSilvaYY) [inline ssl/statem/statem_clnt.c:2855]: Is this TODO still valid ;)

Comment (paulidale) [reply ssl/statem/statem_clnt.c:2855]: Nope.  Will remove.

Comment (paulidale) [reply ssl/statem/statem_clnt.c:2855]: Nope, it's removed in #15539

File: crypto/asn1/p5_pbev2.c [hunk 2]
```diff
@@ -66,11 +66,11 @@ X509_ALGOR *PKCS5_pbe2_set_iv_ex(const EVP_CIPHER *cipher, int iter,
...
     /* Create random IV */
-    if (EVP_CIPHER_iv_length(cipher)) {
+    ivlen = EVP_CIPHER_iv_length(cipher);
+    if (ivlen > 0) {
         if (aiv)
-            memcpy(iv, aiv, EVP_CIPHER_iv_length(cipher));
-        else if (RAND_bytes_ex(libctx, iv, EVP_CIPHER_iv_length(cipher),
-                               0) <= 0)
+            memcpy(iv, aiv, ivlen);
+        else if (RAND_bytes_ex(libctx, iv, ivlen, 0) <= 0)
```
Comment (slontis) [inline crypto/asn1/p5_pbev2.c:73]: Why do you cast sometimes and not others in this PR?

Comment (slontis) [reply crypto/asn1/p5_pbev2.c:73]: See crypto/ec/ecp_s390x_nistp.c

Comment (paulidale) [reply crypto/asn1/p5_pbev2.c:73]: I got sick of adding the casts in :)

**Review by slontis — APPROVED** [review]

File: crypto/asn1/p5_pbev2.c (hunk 2 above)
Comment (slontis) [reply crypto/asn1/p5_pbev2.c:73]: Fix that one case if you like

File: crypto/rsa/rsa_oaep.c [hunk 3]
```diff
@@ -77,6 +77,10 @@ int ossl_rsa_padding_add_PKCS1_OAEP_mgf1_ex(OSSL_LIB_CTX *libctx,
         mgf1md = md;
 
     mdlen = EVP_MD_size(md);
+    if (mdlen < 0) {
```
Comment (t8m) [inline crypto/rsa/rsa_oaep.c:80]: `<= 0` please.

File: crypto/rsa/rsa_pk1.c [hunk 4]
```diff
@@ -138,7 +138,7 @@ int ossl_rsa_padding_add_PKCS1_type_2_ex(OSSL_LIB_CTX *libctx, unsigned char *to
     /* pad out with non-zero random data */
     j = tlen - 3 - flen;
 
-    if (RAND_bytes_ex(libctx, p, j, 0) <= 0)
+    if (j < 0 || RAND_bytes_ex(libctx, p, j, 0) <= 0)
```
Comment (t8m) [inline crypto/rsa/rsa_pk1.c:141]: This `j<0` is impossible condition given the condition above at line 128 (RSA_PKCS1_PADDING_SIZE==11). As Coverity might flag it please do not add it.

Comment (t8m) [reply crypto/rsa/rsa_pk1.c:141]: Hmm, and there should actually be check above for flen > 0 for correctness sake.

**Review by t8m — CHANGES_REQUESTED** [review]

File: crypto/rsa/rsa_oaep.c (hunk 3 above)
Comment (paulidale) [reply crypto/rsa/rsa_oaep.c]: Fixed

File: crypto/rsa/rsa_pk1.c (hunk 4 above)
Comment (paulidale) [reply crypto/rsa/rsa_pk1.c]: done

**Review by t8m — APPROVED** [review]

File: crypto/rsa/rsa_pk1.c (hunk 4 above)
Comment (slontis) [reply crypto/rsa/rsa_pk1.c]: This assumes that tlen is properly sized?
if flen = 0 => j = tlen - 3.
if flen = 1 => j = tlen - 4
if flen = tlen - 11 => j = 8

Comment (t8m) [reply crypto/rsa/rsa_pk1.c]: There is a condition that enforces tlen >= 11 before.

Comment (paulidale) [reply crypto/rsa/rsa_pk1.c]: More specifically, it raises an error if:
```c
flen > tlen - 11
```
with the added `flen >= 0` check, there is no need to check `j` here.

**Comment by paulidale** [comment]
Merged to master.

**Pull request closed without merging**